contracts/beyondFramework/dca/interfaces/IDcaStrategy.sol

```diff
@@ -0,0 +1,100 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import { IDca } from "./IDca.sol";
+import { IDcaFor } from "./IDcaFor.sol";
+import { IDcaInvesting } from "./IDcaInvesting.sol";
+import { SwapLib } from "../libraries/SwapLib.sol";
+
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/interfaces/IERC20Upgradeable.sol";
+
+interface IDcaStrategy is IDca, IDcaFor, IDcaInvesting {
+    error TooSmallDeposit();
+    error PositionsLimitReached();
+    error NothingToInvest();
+    error NothingToWithdraw();
+    error PortfolioAlreadyWhitelisted();
+    error PortfolioNotFound();
+
+    event Deposit(address indexed sender, uint256 amount, uint256 amountSplit);
+    event Invest(
+        uint256 depositAmountSpent,
+        uint256 bluechipReceived,
+        uint256 investedAt
+    );
+    event Withdraw(
+        address indexed sender,
+        uint256 withdrawnDeposit,
+        uint256 withdrawnBluechip
+    );
+    event StatusChanged(
+        BluechipInvestmentState indexed prevStatus,
+        BluechipInvestmentState indexed newStatus
+    );
+    event PortfolioAdded(address indexed newPortfolio);
+    event PortfolioRemoved(address indexed removedPortfolio);
+
+    struct DcaStrategyInitArgs {
+        DepositFee depositFee;
+        address dcaInvestor;
+        TokenInfo depositTokenInfo;
+        uint256 investmentPeriod;
```

Comment:
minor thing: it took me some time to understand this is the **duration** in wall clock of the investmentPeriod, maybe we should change the name to something like that?